substrate/frame/contracts/src/wasm/runtime.rs

```diff
@@ -851,10 +851,11 @@ impl<'a, E: Ext + 'a> Runtime<'a, E> {
 					value,
 					input_data,
 					flags.contains(CallFlags::ALLOW_REENTRY),
+					flags.contains(CallFlags::READ_ONLY),
 				)
 			},
 			CallType::DelegateCall { code_hash_ptr } => {
-				if flags.contains(CallFlags::ALLOW_REENTRY) {
+				if flags.contains(CallFlags::ALLOW_REENTRY | CallFlags::READ_ONLY) {
```

Comment:
Why this change? `READ_ONLY` is allowed to bet set. This checks if both of them are set.

Reply:
In Ethereum, we have `DelegateCall`, `Call`, or `StaticCall`. There is no combination where `DelegateCall` can change the context to `read-only`. Therefore, I decided to have `delegate_call`, `call (without read_only)`, and `call (with read_only)` as the analogy.

Reply:
You are right, thank you